Assets/Scripts/Model/Cards/Corp/HaarpsichordStudios.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using model.cards.types;
+using model.choices.steal;
+using model.timing;
+
+namespace model.cards.corp
+{
+    public class HaarpsichordStudios : Card
+    {
+        override public string FaceupArt => "haarpsichord-studios";
+        override public string Name => "Haarpsichord Studios";
+        override public Faction Faction => Factions.NBN;
+        override public int InfluenceCost { get { throw new System.Exception("Identities don't have an influence cost"); } }
+        override public ICost PlayCost { get { throw new System.Exception("Identities don't have a play cost"); } }
+        override public IEffect Activation => new HaarpsichordEffect();
+        override public IType Type => new Identity();
+
+        private class HaarpsichordEffect : IEffect
+        {
+            IEnumerable<string> IEffect.Graphics => new string[] { };
+            void IEffect.Observe(IImpactObserver observer, Game game) => observer.NotifyImpact(true, this);
+
+            async Task IEffect.Resolve(Game game)
+            {
+                var mod = new HaarpsichordModifier();
+                game.corp.turn.Started += mod.Reset;
+                game.runner.turn.Started += mod.Reset;
+                game.runner.ModifyStealing(mod);
+                await Task.CompletedTask;
+            }
+        }
+
+        private class HaarpsichordModifier : IStealModifier
+        {
+            public bool AgendaStolenThisTurn = false;
+
+            public void Reset(object sender, ITurn turn) {
+                 AgendaStolenThisTurn = false;
+            }
+
+            IStealOption IStealModifier.Modify(IStealOption option)
+            {
+                if (AgendaStolenThisTurn)
+                {
+                    return new CannotSteal();
+                }
+                else
+                {
+                    return new StealingWhileHaarpsichordWatches(option, this);
```

Comment:
I tried a solution with fewer classes, but the `bool AgendaStolenThisTurn` was created too often, effectively always `false`. So the intermediate structures are important, because they control when state changes and for how long it's tracked. Thanks to them I was able to fix this bug: 5c520c6